pkg/cmd/cli/cmd/env.go

```diff
@@ -371,7 +384,11 @@ func updateEnv(existing []kapi.EnvVar, env []kapi.EnvVar, remove []string) []kap
 		covered.Insert(e.Name)
 		out = append(out, e)
 	}
-	return out
+	envUpdated := true
```

Comment:
instead of using reflection, I'd rather initialize envUpdated to false at the top, and set it to true in the loop where we know we're making changes